lib/services/sf.js

```diff
@@ -529,6 +530,41 @@ function SnowflakeService(connectionConfig, httpClient, config)
   {
     return new OperationRequest(options).validate().executeAsync();
   };
+
+  this.getQueryContextDTO = function () {
+    if(!this.qcc){
+      return;
+    }
+    return this.qcc.getQueryContextDTO();
+  };
+
+  this.deserializeQueryContext = function (data) {
+    if(!this.qcc){
+      return;
+    }
+    this.qcc.deserializeQueryContext(data);
+  };
+
+  this.clearCache = function () {
+    if(!this.qcc){
+      return;
+    }
+    this.qcc.clearCache(); 
+  }
+
+  this.initializeQueryContextCache = function (size) {
+     if(!connectionConfig.getDisableQueryContextCache()){
+      this.qcc = new QueryContextCache(size);
+    }
+  }
+
+  // testing purpose
```

Comment:
do we need this function? cannot we write the tests without it?

Reply:
Yes, we should. Except for this, we cannot access the query context cache in the Snowflake service after we complete it.